src/renderer/entities/wallet/lib/wallet-utils.ts

```diff
@@ -118,21 +119,20 @@ function getWalletFilteredAccounts(
 ): Wallet | undefined {
   if (!predicates.walletFn && !predicates.accountFn) return undefined;
 
-  return wallets.reduce<Wallet | undefined>((acc, wallet) => {
-    if (acc) return acc;
+  const wallet = wallets.find((wallet) => {
+    return !predicates.walletFn || predicates.walletFn(wallet);
```

Comment:
I think `!predicates.walletFn` should be checked before, and we shouldn't even do the `.find` in case there's no wallet predicate. Something like

```
let walletsToSearch = wallets

if(!predicates.walletFn){
  wallet = wallets.find((wallet) => { ... }
}
```

Reply:
I am not sure it's needed for wallet, because we need only one wallet, and if walletFn is empty with `find` we get only the first wallet and calculation will be stoped. But I implement such approach for accounts where we use `filter`

Reply:
> I am not sure it's needed for wallet, because we need only one wallet, and if walletFn is empty with find we get only the first wallet and calculation will be stoped.

Ah, that's why I think we need a comment for this function, since it's not clear what we really want to do.
From my understanding of this function originally, you would call it with a `walletFn` and `accountFn` and would get back a wallet that is truthy when you run both functions. Now, if there's no `walletFn` you would expect the function to return a wallet and account that fullfills only the `accountFn`. However, with this implementation, as you said, the first wallet would be selected, and then, if the `accountFn` isn't truthy for any account of this first wallet, the function will return `undefined` although there might be other wallets with an account that would match.

Said differently if I have the following
```ts
const wallets = [
  { 
    id: 1,
    accounts: [
      { accountId: 1 }
    ]
  },
  {
    id: 2,
    accounts: [
    { accountId : 2 }
    ]
  }
]
```

and you call `getWalletFilteredAccounts(wallets, {accountFn: (a) => a.accountId === 2})` then your function will return `undefiened` although it should have returned `{ id: 2, accounts: [ { accountId : 2 } ] }` isn't it?